### PR TITLE
feat: new message for authenticate headers 

### PR DESCRIPTION
**Pull request by Suduck**
# What? 

This PR includes a new message between kernel <> Unity to ask for signed headers. 

This new message has been created to be even more generic than the previous one, this way we can send lots of headers request, and then we can identify back in unity from which request the headers are

# Why? 

This has a more generic implementation and in the future is someone for another system want headers to authenticate, we can ensure that we are getting the requested headers by checking the method and endpoint

### CODE REVIEW

**Review by menduz — DISMISSED** [review]
Code looks hood. Think of adding optional signed headers as per comment

File: packages/unity-interface/BrowserInterface.ts [hunk 1]
```diff
@@ -617,6 +619,15 @@ export class BrowserInterface {
     getUnityInstance().SendBuilderCatalogHeaders(headers)
   }
 
+  public RequestHeaderForEndpoint(data: { method: string; url: string }) {
```
[the remark this answers is not in the document]
Comment (Suduck) [reply packages/unity-interface/BrowserInterface.ts:622]: I will add a generic way to ask headers for other things but unfortunately the builder API does not sign the headers as other systems, as you can see in the image they use a different way. 

I will add the message so if we need it in the future, we already have a way to do it!

image

[the remark this answers is not in the document]
Comment (Suduck) [reply packages/unity-interface/BrowserInterface.ts]: Added!

File: packages/unity-interface/BrowserInterface.ts [hunk 2]
```diff
@@ -613,6 +619,25 @@ export class BrowserInterface {
     getUnityInstance().SendBuilderCatalogHeaders(headers)
   }
 
+  public RequestSignedHeaderForBuilder(data: { method: string; url: string }) {
```
Comment (menduz) [inline packages/unity-interface/BrowserInterface.ts:622]:
```suggestion
  //Note: This message is deprecated and should be deleted in the future.
  //      It is here until the Builder API is stabilized and uses the same signedFetch method as the rest of the platform
  public RequestSignedHeaderForBuilder(data: { method: string; url: string }) {
```

File: packages/unity-interface/BrowserInterface.ts [hunk 3]
```diff
@@ -613,6 +619,25 @@ export class BrowserInterface {
...
+    const identity = getCurrentIdentity(store.getState())
+
+    const headers: Record<string, string> = identity
+      ? BuilderServerAPIManager.authorize(identity, data.method, data.url)
+      : {}
+    getUnityInstance().SendHeaders(data.url, headers)
+  }
+
+  public RequestSignedHeader(data: { method: string; url: string; metadata: Record<string, any> }) {
```
Comment (menduz) [inline packages/unity-interface/BrowserInterface.ts:631]:
```suggestion
  //Note: This message is deprecated and should be deleted in the future.
  //      It is here until the Builder API is stabilized and uses the same signedFetch method as the rest of the platform
  public RequestSignedHeader(data: { method: string; url: string; metadata: Record<string, any> }) {
```

**Review by menduz — APPROVED** [review]

**Pull request merged**